### ISSUE TITLE
Shadow Updates And Normal Maps

### ISSUE THREAD

**Issue by titansoftime** [mrdoob/three.js#9076]
##### Description of the problem

Nice work on the shadow updates. I notice that the acne is gone now on meshes that both cast and receive shadows, Very nice!

One thing I have noticed though is that materials that are covered in shadow lose the normal map effect making for kinda odd scenes. I'm assuming this is from the absence of Directional/Point or Spot Light instead of adding black like it did previously.

![shadows](https://cloud.githubusercontent.com/assets/6549763/15798073/4280c03c-29f2-11e6-9b88-e381e0abdbcd.png)
##### Three.js version
- [ ] Dev
- [x ] r77
- [ ] ...
##### Browser
- [x] All of them
- [ ] Chrome
- [ ] Firefox
- [ ] Internet Explorer
##### OS
- [x] All of them
- [ ] Windows
- [ ] Linux
- [ ] Android
- [ ] IOS

**Comment by mrdoob** [comment]
How did that scene look previously?

**Comment by titansoftime** [comment]
From r72.

![shadows72](https://cloud.githubusercontent.com/assets/6549763/15798188/3147f692-29f6-11e6-8d42-0da64ee45b2a.png)

**Comment by mrdoob** [comment]
Ah, I see... Well, I think you can kind of fake that by using a `HemisphereLight` instead of a `AmbientLight`.

**Comment by WestLangley** [comment]
>I think you can kind of fake that by using a HemisphereLight instead of a AmbientLight.

You can. But since the surface of interest in this example is horizontal, you may find it preferable to tilt the hemisphere light a bit.
```javascript
// ambient
var light = new THREE.HemisphereLight( 0xffffff, 0x000000 );
light.position.set( 1, 1, 0 ).normalize(); // tilt
scene.add( light );
```

**Comment by mrdoob** [comment]
Yes, the `HemisphereLight` should match the `DirectionalLight` direction.

**Comment by WestLangley** [comment]
Agreed.

**Comment by titansoftime** [comment]
I gave it a shot. It looks better. It will take a bit to get the lighting balanced with the additional light, dropping the ambient made the scene look too dark.

r77
![hemi77](https://cloud.githubusercontent.com/assets/6549763/15803167/ff2417ea-2a95-11e6-8c8c-68330592f963.jpg)


Thanks for the workaround =]

**Comment by titansoftime** [comment]
I am reopening this issue due to the workaround while somewhat effective, not really being ideal. 

The additional light comes at a performance cost and getting the normal maps under the shadow and the rest of the scene to match their look from before the shadow refactor is tedious (especially when you have roughly 40 different environments to deal with).

I've experienced similar issues with normal maps (without shadows) where using a darker DirectionalLight color or reduced light Intensity mitigating normal map lighting resulting in ugly flat scenes. Is this intentional? It would be nice to allow for darker scenes to benefit from normal maps.

I would work on this myself but I have limited experience with writhing shaders.
